openapi/src/resources/gcp_ai_notebook_instance.yaml

```diff
@@ -5,7 +5,7 @@ components:
         AI Notebook instance properties included in post-creation, get, and update. Others must be
         retrieved from GCP using the specification.
       type: object
-      required: [projectId, location, instanceId]
+      required: [projectId, location, instanceId, machineType]
```

Comment:
Making this required will not be upward compatible. You need to be able to handle defaulting the machine type.

Reply:
I took a look at the code. IIUC, the existing API requires a machine type to be specified on create. There is no defaulting. And the machine type is currently not update-able. So my initial comment about making the field required is nonsense.

I think that we will want to backfill from adding the machine type and accelerator to the notebook attributes. That can be done as you suggest, by reading the machineType from the cloud platform and recording it in the attributes.